subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -610,6 +618,56 @@ static int tls_opt_sec_tag_list_get(struct net_context *context,
 	return 0;
 }
 
+static int tls_opt_hostname_set(struct net_context *context,
+				const void *optval, socklen_t optlen)
+{
+	ARG_UNUSED(optlen);
+
+	if (!context || !context->tls) {
+		return -EBADF;
+	}
+
+	if (!optval) {
+		return -EFAULT;
+	}
+
+	if (mbedtls_ssl_set_hostname(&context->tls->ssl, optval) != 0) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int tls_opt_hostname_get(struct net_context *context,
+				void *optval, socklen_t *optlen)
+{
+	char *buf = optval;
+
+	if (!context || !context->tls) {
+		return -EBADF;
+	}
+
+	if (!optval || !optlen) {
+		return -EFAULT;
+	}
+
+	if (*optlen == 0) {
```

Comment:
Should we also prevent buffer length 1 here as it would make not much sense because we place \0 at the end?